src/base/RegexPattern.h

```diff
@@ -9,10 +9,15 @@
 #ifndef SQUID_SRC_BASE_REGEXPATTERN_H
 #define SQUID_SRC_BASE_REGEXPATTERN_H
 
-#include "compat/GnuRegex.h"
 #include "mem/forward.h"
 #include "sbuf/SBuf.h"
 
+#if HAVE_REGEX_H
+// regex.h requires size_t from sys/types.h
+#include <sys/types.h>
```

Comment:
Please remove these: regex.h does not require manual sys/types.h inclusion since POSIX 1003.1-2001. See https://sourceware.org/bugzilla/show_bug.cgi?id=2534

Modern regex.h includes the necessary dependencies itself as needed, of course. For example: https://codebrowser.dev/glibc/glibc/posix/regex.h.html#23 . If we are modernizing based on recent code, we should take advantage of improvements made since 1996 instead of increasing code pollution with poorly duplicated sys/types.h `#include` statements.

Reply:
Done